client/object_put.go

```diff
@@ -171,7 +173,10 @@ func (x *ObjectWriter) WritePayloadChunk(chunk []byte) bool {
 func (x *ObjectWriter) Close() (*ResObjectPut, error) {
 	defer x.cancelCtxStream()
 
-	if x.ctxCall.err != nil {
+	// Ignore io.EOF error, because it is expected error for client-side
+	// stream termination by the server. E.g. when stream contains invalid
+	// message. Server returns an error in response message (in status).
+	if x.ctxCall.err != nil && !errors.Is(x.ctxCall.err, io.EOF) {
```

Comment:
So when server has sent status response and closed the stream from its side, then client recvs `io.EOF` on request write, but still can read the status response. Am I correct?

Reply:
Yes. In terms of gRPC, `client.Send(message)` returns `io.EOF`, and then `client.CloseAndRecv()` returns valid response with expected error status inside.